app/src/main/java/com/google/samples/apps/sunflower/compose/home/HomeScreen.kt

```diff
@@ -0,0 +1,163 @@
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun HomeScreen(onPlantClick: (Plant) -> Unit, onPageChange: (SunflowerPage) -> Unit) {
+    val pagerState = rememberPagerState()
+
+    LaunchedEffect(pagerState.currentPage) {
+        onPageChange(SunflowerPage.values()[pagerState.currentPage])
+    }
+
+    Column(Modifier.nestedScroll(rememberNestedScrollInteropConnection())) {
+        val coroutineScope = rememberCoroutineScope()
+
+        // Tab Row
+        TabRow(selectedTabIndex = pagerState.currentPage) {
+            SunflowerPage.values().forEachIndexed { index, page ->
+                val title = stringResource(id = page.titleResId)
+                Tab(
+                    selected = pagerState.currentPage == index,
+                    onClick = { coroutineScope.launch { pagerState.scrollToPage(index) } },
+                    text = { Text(text = title) },
+                    icon = {
+                        Icon(
+                            painter = painterResource(id = page.drawableResId),
+                            contentDescription = title
+                        )
+                    },
+                    unselectedContentColor = MaterialTheme.colors.primaryVariant,
+                    selectedContentColor = MaterialTheme.colors.secondary,
+                )
+            }
+        }
+
+        // Pages
+        HorizontalPager(
+            pageCount = SunflowerPage.values().size,
+            state = pagerState,
+            verticalAlignment = Alignment.Top
+        ) { index ->
+            when (SunflowerPage.values()[index]) {
+                SunflowerPage.MY_GARDEN -> {
+                    GardenScreen(
+                        Modifier.fillMaxSize(),
+                        onAddPlantClick = {
+                            coroutineScope.launch {
+                                pagerState.scrollToPage(SunflowerPage.PLANT_LIST.ordinal)
+                            }
+                        }, onPlantClick = {
+                            onPlantClick(it.plant)
+                        })
+                }
+                SunflowerPage.PLANT_LIST -> {
+                    PlantListScreen(
+                        onPlantClick = onPlantClick,
+                        Modifier.fillMaxSize(),
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun HomeTopAppBar(pagerState: PagerState, onFilterClick: () -> Unit) {
+    TopAppBar(
+        title = {
+            Row(
+                Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Center,
+            ) {
+                Text(
+                    text = stringResource(id = R.string.app_name)
+                )
+            }
+        },
+        Modifier.statusBarsPadding(),
+        actions = {
+            if (pagerState.currentPage == SunflowerPage.PLANT_LIST.ordinal) {
```

Comment:
passing the entire pagerState will make this composable recompose on every page change. I know theres only two pages right now so most likely you do want this, but just wanted to double check ?

Reply:
Yes - this is exactly the intended behavior since the actions change depending on what page you are in.